book.md

```diff
@@ -398,15 +398,76 @@ cache, without the processes having rights to read or write to the same memory
 locations.  To understand how these techniques work, one needs to understand
 typical organization and operation of a cache.
 
-### Typical cache architecture
-
-
-\missingcontent{Add a description of how a cache operates - as far as is
-necessary to explain the cache covert channel techniques.}
-
-### Flush + Reload
-
-### Prime + Probe
+### Typical CPU cache architecture
+
+There is a wide variety in
+[CPU cache micro-architecture](https://en.wikipedia.org/wiki/CPU_cache) details,
+but the main characteristics that are important to set up a covert channel tend
+to be similar across most popular implementations.
+
+Caches are small and much faster memories than the main memory that aim to keep
+a copy of the data at the most frequently accessed addresses in main memory. The
```

Comment:
I think "addresses in main memory" is a little ambiguous here - "a copy of the data in main memory" could mean that either the copy or the data is in main memory. How about "main memory addresses" or "main memory locations"?

Reply:
I decided to go for the suggested "main memory addresses".
I'm assuming some of the confusion may come from different people having a different mental model of what addresses are. Are they pointing to location of the abstract space of all memory locations in a running program (i.e. close to being a mathematical model)? Or are they the physical locations in DRAM chips? Or something else?
I hope the paragraph now becomes unambiguous enough that we won't have to go into "what is an address really?"....